doc/howto/set-up-ci-with-package-management.md

```diff
@@ -0,0 +1,71 @@
+# How to set up Continuous Integration with Package Management
+
+For many projects published it can be useful to set up a [continuous
+integration](https://en.wikipedia.org/wiki/Continuous_integration) (CI) system.
+This allows to check whether the code builds on other computers, the tests pass
+and helps to assess contributions from other people.
+
+There are numerous CI systems available. In this document we will be using
+GitHub Actions, as it is integrated in GitHub and free to use.
+
+## What are Actions
+
+Projects on GitHub can set up [Actions](https://docs.github.com/en/actions),
+that is workflows defined in YAML files. These definitions are placed in the
+`.github/workflows/` folder in the Git repository.
+
+These workflows consist of the execution of a sequence of actions, like running
+shell commands. It is also possible to use pre-defined actions or even
+user-submitted functionality.
+
+## `setup-dune`
+
+[`setup-dune`](https://github.com/ocaml-dune/setup-dune) is one such custom
+action maintained by the OCaml community to help with setting up builds with
+Dune.
+
+It installs Dune and will attempt to build the source code.
+
+::::{dropdown} `.github/workflow/ci.yaml`
+:icon: file-code
+
+:::{literalinclude} set-up-ci-with-package-management/ci.yaml
+:language: yaml
+:emphasize-lines: 15-18
+:::
+
+We define a job called `build`, which will be run on Ubuntu and macOS (a [list
+of possible
+options](https://docs.github.com/en/actions/reference/runners/github-hosted-runners)
+can be found in the GitHub documentation).
+
+This job first checks out the projects' source code using an action maintained
+by GitHub and then uses the `setup-dune` action to install Dune on the system
+and build the project.
+
+It parametrizes the `setup-dune` action with `automagic` set to `true` which
+instructs the action, amongst other things, to tell Dune to create a lock
+directory. This lock directory will then be used to download, build and install
+the project dependencies before building the project itself.
+
+::::
+
+This configuration file has to be added to the repository and pushed to Github.
```

Comment:
You have used “GitHub” & now “Github” inconsistently. I would prefer the actual casing of “GitHub”. #12994